src/test/java/unit/BqDatasetLifetime.java

```diff
@@ -0,0 +1,227 @@
+    assertEquals(0, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+
+    updateDatasetWithDefaultLifetimes(
+        renamed, "--new-name=" + name, "--default-table-lifetime=" + tableLifetime.toString());
+    defaultLifetimeMap = getDefaultLifetimes(bqDataset);
+    assertEquals(partitionLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.PARTITION));
+    assertEquals(tableLifetime, defaultLifetimeMap.get(DefaultLifetimeEntity.TABLE));
+  }
+
+  /**
+   * Helper method to get the name of the dataset that should be created by the test method (derived
+   * from the passed TestInfo instance).
+   */
+  private String getDatasetName(TestInfo testInfo) {
+    return testInfo.getTestMethod().orElseThrow().getName();
+  }
+
+  /** Creates a BQ Dataset with the given name and any passed lifetime arguments. */
+  private UFBqDataset createDatasetWithLDefaultLifetimes(
+      String name, String... defaultLifetimeArguments) throws JsonProcessingException {
+
+    String datasetId = randomDatasetId();
+
+    List<String> arguments =
+        new ArrayList<>(
+            Arrays.asList(
+                "resource", "create", "bq-dataset", "--name=" + name, "--dataset-id=" + datasetId));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+
+    return TestCommand.runAndParseCommandExpectSuccess(
+        UFBqDataset.class, arguments.toArray(new String[0]));
+  }
+
+  /** Updates a dataset with a given name using the passed arguments. */
+  private void updateDatasetWithDefaultLifetimes(String name, String... defaultLifetimeArguments) {
+    List<String> arguments =
+        new ArrayList<>(Arrays.asList("resource", "update", "bq-dataset", "--name=" + name));
+
+    arguments.addAll(Arrays.asList(defaultLifetimeArguments));
+    TestCommand.runCommandExpectSuccess(arguments.toArray(new String[0]));
+  }
+
+  /** Enum to use as a key for a map of entity to lifetime values. */
+  private enum DefaultLifetimeEntity {
+    PARTITION,
+    TABLE
+  }
+
+  /**
+   * Use the Google BQ Client to get the configured default lifetime for the passed dataset
+   * descriptor and return it as a map of entity type (partition or table) to its configured default
+   * lifetime in seconds.
+   */
+  private Map<DefaultLifetimeEntity, Integer> getDefaultLifetimes(UFBqDataset bqDataset)
+      throws IOException {
+
+    DatasetId datasetId = DatasetId.of(bqDataset.projectId, bqDataset.datasetId);
+
+    Dataset dataset =
+        ExternalBQDatasets.getBQClient(workspaceCreator.getCredentialsWithCloudPlatformScope())
+            .getDataset(datasetId);
+
+    assertNotNull(dataset, "looking up dataset via BQ API succeeded");
+
+    // These methods return null if lifetime is not set, if they are set they need to be converted
+    // to integers representing seconds to be returned in the map, with 0 representing no lifetime
+    // set.  Both of these methods return milliseconds.
```

Comment:
Do you need to convert `null` to `0` here? I'm not sure what you get from that over just `assertNull` in your tests.